presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/ResourceGroup.java

```diff
@@ -449,6 +537,26 @@ protected void internalRefreshStats()
         }
     }
 
+    protected void internalGenerateCpuQuota(long elapsedSeconds)
+    {
+        checkState(Thread.holdsLock(root), "Must hold lock to generate cpu quota");
+        synchronized (root) {
+            long newQuota;
+            try {
+                newQuota = Math.multiplyExact(elapsedSeconds, cpuQuotaGenerationMillisPerSecond);
+            }
+            catch (ArithmeticException e) {
+                newQuota = Long.MAX_VALUE;
+            }
+            // This will never overflow, since cpuUsageMillis is >= 0 and |Long.MAX_VALUE| < |Long.MIN_VALUE|
+            cpuUsageMillis -= newQuota;
```

Comment:
I'd just move this to the try and use subtract exact so you don't have to explain it.  Also it will be safer when you change this code.